File: 2021/template/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func playOne(input []string) {
	fmt.Println("Not Yet Implemented")
}

func playTwo(intput []string) {
	fmt.Println("Not Yet Implemented")
}

func main() {
	buffer := 1
	var err error
	if len(os.Args) > 1 {
		if buffer, err = strconv.Atoi(os.Args[1]); err != nil {
			log.Fatal(("Could not convert arg to number: " + os.Args[1]))
		}
	}

	input := make([]string, 0)
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		positions = append(positions, scanner.Text())
	}

	switch buffer {
	case 1:
		playOne(positions)
	case 2:
		playTwo(positions)
	}
}

File: 2021/template/main.cc
#include <iostream>
#include <string>
#include <vector>

void part1() {
  std::cout << "Not Yet Implemented" << std::endl;
}

void part2() {
  std::cout << "Not Yet Implemented" << std::endl;
}

int main(int argc, char** argv) {
  int buffer = 1;
	if (argc > 1) {
    buffer = std::atoi(argv[1]);
	}

  if (buffer == 1) {
    part1();
  } else if (buffer == 2) {
    part2();
  }

  return 0;
}

